lib/snippets/kubernetes-deploy

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env ruby
+
+# Usage: kubernetes-deploy <app's namespace> <kube context>
+
+# Prerequisites:
+#  - kubectl 1.5.1+ binary must be available in the shipit machine's path
+#  - ENV['KUBECONFIG'] must point to a valid kubeconfig file that includes all the contexts you want to deploy to
+#  - ENV['GOOGLE_APPLICATION_CREDENTIALS'] must point to the credentials for an authenticated service account if your user's auth provider is gcp
+
+# Optionally, the following variables can be used to override script defaults:
+#  - ENV['K8S_TEMPLATE_FOLDER']: Location of Kubernetes files to deploy. Default is config/deploy/#{environment}.
+
+require 'open3'
+require 'securerandom'
+require 'erb'
+require 'json'
+require 'yaml'
+require 'shellwords'
+require 'tempfile'
+require 'logger'
+
+class KubernetesDeploy
+  class FatalDeploymentError < StandardError; end
+
+  def initialize(namespace:, environment:, current_sha:, template_folder: nil, context:)
+    @namespace = namespace
+    @context = context
+    @current_sha = current_sha
+    @template_path = File.expand_path('./' + (template_folder || "config/deploy/#{environment}"))
+    # Max length of podname is only 63chars so try to save some room by truncating sha to 8 chars
+    @id = current_sha[0...8] + "-#{SecureRandom.hex(4)}" if current_sha
+  end
+
+  def run
+    validate_configuration
+    set_kubectl_context
+    validate_namespace
+    apply_all_templates
+  rescue FatalDeploymentError => error
+    logger.fatal(error.message)
+    exit 1
+  end
+
+  def template_variables
+    {
+      'current_sha' => @current_sha,
+      'deployment_id' => @id,
+    }
+  end
+
+  private
+
+  def validate_configuration
+    errors = []
+    if ENV["KUBECONFIG"].nil? || !File.file?(ENV["KUBECONFIG"])
+      errors << "Kube config not found at #{ENV["KUBECONFIG"]}"
+    end
+
+    if @current_sha.nil?  || @current_sha.empty?
+      errors << "Current SHA must be specified"
+    end
+
+    if !File.directory?(@template_path)
+      errors << "Template path #{@template_path} doesn't exist"
+    elsif Dir.entries(@template_path).select {|file| file =~ /\.yml(.erb)?$/}.size.zero?
```

Comment:
The `.` should be escaped in this regex: `\.erb`.

Also: `.size.zero?` -> `empty?`